### PR TITLE
add link validation

### PR DESCRIPTION
**Pull request by afeld**
On the fence about whether we need to test the private *and* public versions of the site...strong feelings one way or another? Basically, the tradeoff is test running time vs. making sure the public Hub experience is :sparkles: 

Going to play around with parallelizing this + running tests, too. You know, for funsies.

### CODE REVIEW

**Comment by mbland** [comment]
Going to play with this now. Definitely shouldn't run the test for the internal version on Travis.

**Comment by mbland** [comment]
Just tried this again after merging with the current `master`:

```shell
$ ./go validate
Running ["ScriptCheck", "LinkCheck", "ImageCheck"] checks on ./_site on *.html... 

/usr/local/var/rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/html-proofer-2.3.0/lib/html/proofer/checkable.rb:13:in `instance_variable_set': `@b.' is not allowed
 as an instance variable name (NameError)
        from /usr/local/var/rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/html-proofer-2.3.0/lib/html/proofer/checkable.rb:13:in `block in initialize'
```

**Comment by afeld** [comment]
@mbland Just merged master, and the link validation works locally and on Travis (hooray!).

File: go [hunk 1]
```diff
@@ -95,9 +95,31 @@ def build
...
+def validate
+  build
+  validate_public
+  validate_private
+end
+
 def ci_build
   test
   build
+  validate_public
```
Comment (afeld) [inline go]: Maybe we should take this out or rescue the exception so that we can merge this without all of our builds breaking until we fix them...

**Comment by afeld** [comment]
*Weeeeeirdd....*

```
$ bundle exec ./go validate_public
Running ["ImageCheck", "ScriptCheck", "LinkCheck"] checks on ./_site_public on *.html...


Ran on 327 files!


- ./_site_public/hub/101/vagrant-and-ansible/index.html
  *  internally linking to deploy/ansible/playbook.yml, which does not exist (line 224)
...
/Users/aidanfeldman/.rvm/gems/ruby-2.2.3/gems/html-proofer-2.3.0/lib/html/proofer.rb:186:in `print_failed_tests': HTML-Proofer found 35 failures! (RuntimeError)
	from /Users/aidanfeldman/.rvm/gems/ruby-2.2.3/gems/html-proofer-2.3.0/lib/html/proofer.rb:88:in `run'
	from ./go:65:in `block in <main>'
	from /Users/aidanfeldman/.rvm/gems/ruby-2.2.3/gems/go_script-0.1.0/lib/go_script/go.rb:24:in `call'
	from /Users/aidanfeldman/.rvm/gems/ruby-2.2.3/gems/go_script-0.1.0/lib/go_script/go.rb:24:in `block in def_command'
	from /Users/aidanfeldman/.rvm/gems/ruby-2.2.3/gems/go_script-0.1.0/lib/go_script/go.rb:34:in `execute_command'
	from ./go:124:in `<main>'
$ bundle exec ./go validate_private
Running ["ScriptCheck", "LinkCheck", "ImageCheck"] checks on ./_site on *.html...


/Users/aidanfeldman/.rvm/gems/ruby-2.2.3/gems/html-proofer-2.3.0/lib/html/proofer/checkable.rb:13:in `instance_variable_set': `@b.' is not allowed as an instance variable name (NameError)
	from /Users/aidanfeldman/.rvm/gems/ruby-2.2.3/gems/html-proofer-2.3.0/lib/html/proofer/checkable.rb:13:in `block in initialize'
	from /Users/aidanfeldman/.rvm/gems/ruby-2.2.3/gems/html-proofer-2.3.0/lib/html/proofer/checkable.rb:12:in `each_pair'
	from /Users/aidanfeldman/.rvm/gems/ruby-2.2.3/gems/html-proofer-2.3.0/lib/html/proofer/checkable.rb:12:in `initialize'
	from /Users/aidanfeldman/.rvm/gems/ruby-2.2.3/gems/html-proofer-2.3.0/lib/html/proofer/checks/images.rb:32:in `new'
	from /Users/aidanfeldman/.rvm/gems/ruby-2.2.3/gems/html-proofer-2.3.0/lib/html/proofer/checks/images.rb:32:in `block in run'
	from /Users/aidanfeldman/.rvm/gems/ruby-2.2.3/gems/nokogiri-1.6.6.2/lib/nokogiri/xml/node_set.rb:187:in `block in each'
	from /Users/aidanfeldman/.rvm/gems/ruby-2.2.3/gems/nokogiri-1.6.6.2/lib/nokogiri/xml/node_set.rb:186:in `upto'
	from /Users/aidanfeldman/.rvm/gems/ruby-2.2.3/gems/nokogiri-1.6.6.2/lib/nokogiri/xml/node_set.rb:186:in `each'
	from /Users/aidanfeldman/.rvm/gems/ruby-2.2.3/gems/html-proofer-2.3.0/lib/html/proofer/checks/images.rb:31:in `run'
	from /Users/aidanfeldman/.rvm/gems/ruby-2.2.3/gems/html-proofer-2.3.0/lib/html/proofer.rb:128:in `block (2 levels) in check_files_for_internal_woes'
	from /Users/aidanfeldman/.rvm/gems/ruby-2.2.3/gems/html-proofer-2.3.0/lib/html/proofer.rb:125:in `each'
	from /Users/aidanfeldman/.rvm/gems/ruby-2.2.3/gems/html-proofer-2.3.0/lib/html/proofer.rb:125:in `block in check_files_for_internal_woes'
	from /Users/aidanfeldman/.rvm/gems/ruby-2.2.3/gems/parallel-1.6.1/lib/parallel.rb:408:in `call'
	from /Users/aidanfeldman/.rvm/gems/ruby-2.2.3/gems/parallel-1.6.1/lib/parallel.rb:408:in `call_with_index'
	from /Users/aidanfeldman/.rvm/gems/ruby-2.2.3/gems/parallel-1.6.1/lib/parallel.rb:379:in `process_incoming_jobs'
	from /Users/aidanfeldman/.rvm/gems/ruby-2.2.3/gems/parallel-1.6.1/lib/parallel.rb:361:in `block in worker'
	from /Users/aidanfeldman/.rvm/gems/ruby-2.2.3/gems/parallel-1.6.1/lib/parallel.rb:354:in `fork'
	from /Users/aidanfeldman/.rvm/gems/ruby-2.2.3/gems/parallel-1.6.1/lib/parallel.rb:354:in `worker'
	from /Users/aidanfeldman/.rvm/gems/ruby-2.2.3/gems/parallel-1.6.1/lib/parallel.rb:345:in `block in create_workers'
	from /Users/aidanfeldman/.rvm/gems/ruby-2.2.3/gems/parallel-1.6.1/lib/parallel.rb:344:in `each'
	from /Users/aidanfeldman/.rvm/gems/ruby-2.2.3/gems/parallel-1.6.1/lib/parallel.rb:344:in `create_workers'
	from /Users/aidanfeldman/.rvm/gems/ruby-2.2.3/gems/parallel-1.6.1/lib/parallel.rb:301:in `work_in_processes'
	from /Users/aidanfeldman/.rvm/gems/ruby-2.2.3/gems/parallel-1.6.1/lib/parallel.rb:232:in `map'
	from /Users/aidanfeldman/.rvm/gems/ruby-2.2.3/gems/html-proofer-2.3.0/lib/html/proofer.rb:121:in `check_files_for_internal_woes'
	from /Users/aidanfeldman/.rvm/gems/ruby-2.2.3/gems/html-proofer-2.3.0/lib/html/proofer.rb:107:in `check_directory_of_files'
	from /Users/aidanfeldman/.rvm/gems/ruby-2.2.3/gems/html-proofer-2.3.0/lib/html/proofer.rb:82:in `run'
	from ./go:70:in `block in <main>'
	from /Users/aidanfeldman/.rvm/gems/ruby-2.2.3/gems/go_script-0.1.0/lib/go_script/go.rb:24:in `call'
	from /Users/aidanfeldman/.rvm/gems/ruby-2.2.3/gems/go_script-0.1.0/lib/go_script/go.rb:24:in `block in def_command'
	from /Users/aidanfeldman/.rvm/gems/ruby-2.2.3/gems/go_script-0.1.0/lib/go_script/go.rb:34:in `execute_command'
	from ./go:124:in `<main>'

**Comment by mbland** [comment]
That's what I was seeing until I made a fresh clone...

...also, you don't need to `bundle exec ./go`. It requires `bundler/setup`, so it's good to go.

**Pull request merged**